src/sage/rings/ring_extension_element.pyx

```diff
@@ -354,7 +379,12 @@ cdef class RingExtensionElement(CommutativeAlgebraElement):
             f = parent._backend_defining_morphism
             base = f.domain()
             ring = f.codomain()
+            parent_ring = self._backend.parent()
             if ring.has_coerce_map_from(base) and are_equal_morphisms(f, None):
+                if is_PolynomialQuotientRing(parent_ring):
+                    if base.has_coerce_map_from(parent_ring.base_ring()):
+                        if self._backend.lift().degree() <= 0:
+                            return parent.base()(self._backend.list()[0])
```

Comment:
I would say that it is not the correct place for this piece of code.

You should instead make the following work:
```
sage: Fq.<a> = GF(5^2)
sage: A.<T> = Fq[]
sage: K.<b> = Fq.extension(T^2 + a)
sage: Fq(K(a))
Traceback (most recent call last):
...
TypeError: unable to coerce <class 'sage.rings.polynomial.polynomial_quotient_ring.PolynomialQuotientRing_field_with_category.element_class'>
```

Reply:
While that type of coercion would be nice to have, the one we want in the Drinfeld module implementation is a bit different since we need to coerce an element from a ring extension over a base to a base element, so it would be something along the lines of 
```
sage: Fq.<a> = GF(5^2)
sage: A.<T> = Fq[]
sage: K.<b> = Fq.extension(T^2 + a)
sage: L = K.over(Fq)
sage: Fq(L(a))
Traceback (most recent call last):
...
TypeError: unable to coerce <class 'sage.rings.ring_extension_element.RingExtensionElement'>

This is necessary for methods like `invert` which need to map coefficients into $\mathbb{F}_q$, and that is already handled by the `in_base` method for `FiniteField` classes. It seems to me a bit odd to have to decide between these two different coercion styles into the base for ring extensions. One other option could be to not use RingExtensions for Drinfeld modules over univariate quotients, but I think that could be a bit more complicated to support.

Reply:
> force a user to have to decide between these two different ways of mapping into the base for ring extensions depending on the type of the extension ring

What do you mean?

> Another could be to not use RingExtensions for Drinfeld modules over univariate quotients

Yeah, that would be a bit weird, no? Fundamentally, we used ring extensions to see the ground field $K$ as a $\mathbb F_q[T]$-field. We want to be able to do that without any assumption on $K$. Am I understanding correctly what you mean?